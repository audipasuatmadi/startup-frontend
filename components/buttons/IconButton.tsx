import React from 'react'
import BaseButton from './BaseButton'

export interface IconButtonProps {
  disabled?: boolean;
  href?: string;

  className?: string;
  type?: 'button' | 'submit' | 'reset';

  onClick?: () => void;
  icon:  React.FunctionComponent<React.SVGProps<SVGSVGElement>> | (() => JSX.Element) | JSX.Element;
}

const IconButton = ({disabled, className, icon: Icon, ...otherProps}: IconButtonProps) => {

  const finalStyles = `p-1 rounded outline-none focus:outline-none flex items-center justify-center ${
    disabled
      ? `cursor-not-allowed`
      : `bg-gray-500 bg-opacity-0 hover:bg-opacity-50 transition-all focus:ring-2 ring-brand ring-opacity-50 active:bg-gray-300`
  }`;

  return (
    <BaseButton className={`${finalStyles}${className? ` ${className}`: ''}`} disabled={disabled}>
      {Icon}
    </BaseButton>
  )
}

export default IconButton
