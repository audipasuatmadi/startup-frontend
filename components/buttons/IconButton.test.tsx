import React from 'react'
import { ShallowWrapper, shallow } from 'enzyme';
import IconButton, { IconButtonProps } from './IconButton';
import Image from 'next/image'

const baseProps: IconButtonProps = {
  icon: <Image src="/assets/svgs/ic-search.svg" height={32} width={32} />,
};

describe('icon button rendering tests', () => {
  let wrapper: ShallowWrapper;
  let props: IconButtonProps;

  const renderComponent = (props: IconButtonProps) => <IconButton {...props} />;

  afterEach(() => {
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });

  beforeEach(() => {
    props = baseProps;
  });

  it('renders correctly without props', () => {
    wrapper = shallow(renderComponent(props));
  });

  it('renders correctly as a Link', () => {
    props = { ...props, href: '/' };
    wrapper = shallow(renderComponent(props));
  });

  it('renders correctly when disabled', () => {
    props = { ...props, disabled: true };
    wrapper = shallow(renderComponent(props));
  });

  it('renders correctly as a Link when disabled', () => {
    props = { ...props, disabled: true, href: '/' };
    wrapper = shallow(renderComponent(props));
  });
});
