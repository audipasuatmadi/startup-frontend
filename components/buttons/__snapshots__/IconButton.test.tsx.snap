// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon button rendering tests renders correctly as a Link 1`] = `
<BaseButton
  className="p-1 rounded outline-none focus:outline-none flex items-center justify-center bg-gray-500 bg-opacity-0 hover:bg-opacity-50 transition-all focus:ring-2 ring-brand ring-opacity-50 active:bg-gray-300"
  href="/"
>
  <Image
    height={32}
    src="/assets/svgs/ic-search.svg"
    width={32}
  />
</BaseButton>
`;

exports[`icon button rendering tests renders correctly as a Link when disabled 1`] = `
<BaseButton
  className="p-1 rounded outline-none focus:outline-none flex items-center justify-center cursor-not-allowed"
  disabled={true}
  href="/"
>
  <Image
    height={32}
    src="/assets/svgs/ic-search.svg"
    width={32}
  />
</BaseButton>
`;

exports[`icon button rendering tests renders correctly when disabled 1`] = `
<BaseButton
  className="p-1 rounded outline-none focus:outline-none flex items-center justify-center cursor-not-allowed"
  disabled={true}
>
  <Image
    height={32}
    src="/assets/svgs/ic-search.svg"
    width={32}
  />
</BaseButton>
`;

exports[`icon button rendering tests renders correctly without props 1`] = `
<BaseButton
  className="p-1 rounded outline-none focus:outline-none flex items-center justify-center bg-gray-500 bg-opacity-0 hover:bg-opacity-50 transition-all focus:ring-2 ring-brand ring-opacity-50 active:bg-gray-300"
>
  <Image
    height={32}
    src="/assets/svgs/ic-search.svg"
    width={32}
  />
</BaseButton>
`;
